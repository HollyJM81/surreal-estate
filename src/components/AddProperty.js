import React, { useState } from "react";
import "../styles/add-property.css";
import CurrencyInput from "react-currency-input-field";

const axios = require("axios").default;

const AddProperty = () => {
	const initialState = {
		fields: {
			title: "",
			city: "Manchester",
		},
	};

	const [fields, setFields] = useState(initialState.fields);
	const handleAddProperty = (event) => {
		event.preventDefault();
		console.log(fields);
	};

	const handleFieldChange = (event) => {
		setFields({ ...fields, [event.target.name]: event.target.value });
	};

	return (
		<div className="add-property">
			<h1 className="add-property__title">Add your property details here</h1>
			<form onSubmit={handleAddProperty} className="add-property__form">
				<label htmlFor="title">
					Title
					<input
						id="title"
						type=""
						name="title"
						defaultValue=""
						onChange={handleFieldChange}
					/>
				</label>

				<label htmlFor="type">
					Type
					<select
						id="type"
						name="type"
						value={fields.type}
						onChange={handleFieldChange}
					>
						<option value="Flat">Flat</option>
						<option value="Detached">Detached</option>
						<option value="Semi-Detached">Semi-Detached</option>
						<option value="Terraced">Terraced</option>
						<option value="End of Terrace">End of Terrace</option>
						<option value="Cottage">Cottage</option>
						<option value="Bungalow">Bungalow</option>
					</select>
				</label>

				<label htmlFor="bedrooms">
					Bedrooms
					<select
						id="bedrooms"
						type="number"
						name="bedrooms"
						value={fields.bedrooms}
						onChange={handleFieldChange}
					>
						<option value="1">1</option>
						<option value="2">2</option>
						<option value="3">3</option>
						<option value="4">4</option>
						<option value="5+">5+</option>
					</select>
				</label>

				<label htmlFor="bathrooms">
					Bathrooms
					<select
						id="bathrooms"
						type="number"
						name="bathrooms"
						value={fields.bathrooms}
						onChange={handleFieldChange}
					>
						<option value="1">1</option>
						<option value="2">2</option>
						<option value="3">3</option>
						<option value="4+">4=</option>
					</select>
				</label>

				<label htmlFor="price">
					Price
					<input
						id="price"
						name="price"
						placeholder="Please enter a price"
						// prefix="Â£"
						// defaultValue={1000}
						// decimalsLimit={2}
						value={fields.price}
						onChange={handleFieldChange}
					/>
				</label>

				<label htmlFor="city">
					City
					<select
						id="city"
						name="city"
						value={fields.city}
						onChange={handleFieldChange}
					>
						<option value="Manchester">Manchester</option>
						<option value="Leeds">Leeds</option>
						<option value="Sheffield">Sheffield</option>
						<option value="Liverpool">Liverpool</option>
					</select>
				</label>

				<label htmlFor="email">
					Email
					<input
						id="email"
						type="email"
						name="email"
						defaultValue=""
						onChange={handleFieldChange}
					/>
				</label>
				<button type="submit" className="add-property__submit-btn">
					Add
				</button>
			</form>
		</div>
	);
};

export default AddProperty;
