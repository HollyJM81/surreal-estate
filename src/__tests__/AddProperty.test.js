import { render } from "@testing-library/react";
import AddProperty from "../components/AddProperty";

const mockData = {
	title: "Spacious City Apartment",
	type: "Flat",
	bedrooms: "2",
	bathrooms: "1",
	price: "170000",
	city: "Manchester",
	email: "info@manchesterflats.com",
};

describe("AddProperty", () => {
	it("renders correctly", () => {
		const { asFragment, getByTestId } = render(<AddProperty />);
		expect(asFragment()).toMatchSnapshot();
		expect(getByTestId("add-property-form")).toBeInTheDocument();
	});
});
