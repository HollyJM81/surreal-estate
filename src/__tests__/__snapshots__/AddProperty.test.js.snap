// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AddProperty renders correctly 1`] = `
<DocumentFragment>
  <div
    class="add-property"
  >
    <h1
      class="add-property__title"
    >
      Add your property details here
    </h1>
    <form
      class="add-property__form"
      data-testid="add-property-form"
    >
      <label
        for="title"
      >
        Title
        <input
          id="title"
          name="title"
          type=""
          value=""
        />
      </label>
      <label
        for="type"
      >
        Type
        <select
          id="type"
          name="type"
        >
          <option
            value="Flat"
          >
            Flat
          </option>
          <option
            value="Detached"
          >
            Detached
          </option>
          <option
            value="Semi-Detached"
          >
            Semi-Detached
          </option>
          <option
            value="Terraced"
          >
            Terraced
          </option>
          <option
            value="End of Terrace"
          >
            End of Terrace
          </option>
          <option
            value="Cottage"
          >
            Cottage
          </option>
          <option
            value="Bungalow"
          >
            Bungalow
          </option>
        </select>
      </label>
      <label
        for="bedrooms"
      >
        Bedrooms
        <select
          id="bedrooms"
          name="bedrooms"
          type="number"
        >
          <option
            value="1"
          >
            1
          </option>
          <option
            value="2"
          >
            2
          </option>
          <option
            value="3"
          >
            3
          </option>
          <option
            value="4"
          >
            4
          </option>
          <option
            value="5+"
          >
            5+
          </option>
        </select>
      </label>
      <label
        for="bathrooms"
      >
        Bathrooms
        <select
          id="bathrooms"
          name="bathrooms"
          type="number"
        >
          <option
            value="1"
          >
            1
          </option>
          <option
            value="2"
          >
            2
          </option>
          <option
            value="3"
          >
            3
          </option>
          <option
            value="4+"
          >
            4=
          </option>
        </select>
      </label>
      <label
        for="price"
      >
        Price
        <input
          id="price"
          name="price"
          placeholder="Please enter a price"
          value="0"
        />
      </label>
      <label
        for="city"
      >
        City
        <select
          id="city"
          name="city"
        >
          <option
            value="Manchester"
          >
            Manchester
          </option>
          <option
            value="Leeds"
          >
            Leeds
          </option>
          <option
            value="Sheffield"
          >
            Sheffield
          </option>
          <option
            value="Liverpool"
          >
            Liverpool
          </option>
        </select>
      </label>
      <label
        for="email"
      >
        Email
        <input
          id="email"
          name="email"
          type="email"
          value=""
        />
      </label>
      <div
        class="add-property__submit"
      >
        <button
          class="add-property__submit-btn"
          type="submit"
        >
          Add
        </button>
        <div
          class="Alert--error"
          data-testid="alert"
        />
      </div>
    </form>
  </div>
</DocumentFragment>
`;
